### ISSUE TITLE
 Protractor + Storybook: Failed: No element found using locator: By(css selector)

### ISSUE THREAD

**Issue by rowindatas25** [storybookjs/storybook#12554]
**Describe the bug**

I am trying to implement e2e testing of UI components for Storybook with Protractor. Any attempt to use a locator to find an element on the page provides an error in the terminal saying the element cannot be found using a Protractor locator, i.e (by.css, by.id and by.className). However, when running tests involving button text, they pass successfully with no errors.

**To Reproduce**

To reproduce the issue, protractor must be installed within a storybook project using Angular. Run the protractor command to start the test. 

Test fails with this error in the terminal: Failed: No element found using locator: By(css selector)

**Expected behavior**

The expected behavior is to locate an input field on the screen and pass text into the input field to check that the value of that input is equal to a specific value.

**Screenshots**

**Code snippets**

conf.js
```
exports.config = {
  framework: 'jasmine',
  seleniumAddress: 'http://localhost:4444/wd/hub',
  specs: ['spec.js'],
  capabilities: {
    browserName: 'chrome'
  }
}

```
forms-spec.js
```
describe('Contact Form', function() {
    browser.waitForAngularEnabled(false);
    browser.ignoreSynchronization = true;
    it('should allow me to enter a first name', async() => {
    await browser.get('http://localhost:6006/?path=/story/components-contact-form--default');

    let text = element(by.css('input'));
    element(by.css('input')).sendKeys('Rohtas');
    expect(text.getAttribute('value')).toEqual('Rohtas');
      
    });

  })
```

Error:  NoSuchElementError: No element found using locator: By(css selector, input)

**System:**

 System:
    OS: Windows 10 10.0.17763
    CPU: (8) x64 Intel(R) Core(TM) i5-8365U CPU @ 1.60GHz
  Binaries:
    Node: 12.16.3 - C:\Program Files\nodejs\node.EXE
    npm: 6.14.5 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Chrome: 83.0.4103.97
    Edge: Spartan (44.17763.831.0)
  npmPackages:
    @storybook/addon-a11y: ^5.3.19 => 5.3.19
    @storybook/addon-actions: ^5.3.19 => 5.3.19
    @storybook/addon-docs: ^5.3.19 => 5.3.19
    @storybook/addon-jest: ^5.3.19 => 5.3.19
    @storybook/addon-knobs: ^5.3.19 => 5.3.19
    @storybook/addon-links: ^5.3.19 => 5.3.19
    @storybook/addon-notes: ^5.3.19 => 5.3.19
    @storybook/addon-storysource: ^5.3.19 => 5.3.19
    @storybook/addon-viewport: ^5.3.19 => 5.3.19
    @storybook/addons: ^5.3.19 => 5.3.19
    @storybook/angular: ^5.3.19 => 5.3.19
    @storybook/storybook-deployer: ^2.8.6 => 2.8.6

**Additional context**

If I add a className or ID to the input field, and try to locate using either method the same error appears.
However, using buttonText or partialButtonText on a button from the Protractor API works fine. Is there a compatibility issue between Protractor and Storybook I was not aware of.

**Comment by shilman** [comment]
I havne't looked in detail but Storybook has an outer frame (aka "manager") and an inner iframe (aka "preview"). Based on your description I bet you're trying to select from the preview, but protractor's doing the select on the manager and failing.

You can eject the preview using the eject button on the right hand side of storybook's toolbar:

https://storybook.js.org/docs/react/get-started/browse-stories#toolbar

**Comment by rowindatas25** [comment]
Hi Michael, 

Thank you for getting back to me. When I try that the error is still present, please let me know if there's anything else I can provide.

**Comment by rowindatas25** [comment]
Hi Michael,

Was anyone able to look into this? I've also tried running the same test in Cypress and the issue remains the same, storybook's iframe might be what's blocking the locators from being found.

**Comment by shilman** [comment]
Nope, can you share what you tried? This is wrong:

```
    await browser.get('http://localhost:6006/?path=/story/components-contact-form--default');
```

I would expect something like:

```
    await browser.get('http://localhost:6006/iframe.html?id=components-contact-form--default&viewMode=story');
```

Using the URL that the "eject" button gives you (mentioned above ☝️

**Comment by rowindatas25** [comment]
I did try the first time with the link you just posted but the same error appears. However, when I tried that same link in the other automation software cypress, this link did work! 

I may look to switch over to Cypress instead of Protractor as a result but would be able to do some more debugging with protractor to see why it might not be working?

Thank you very much again for your help Michael!
